server/server.js

```diff
@@ -1,15 +1,164 @@
 const express = require("express");
+let cors = require('cors')
 const app = express();
 const port = process.env.PORT || 5000;
+const mysql = require('mysql');
 
+app.use(express.urlencoded({ extended: false }));
+app.use(express.json());
+app.use(cors());
 app.listen(port, () => console.log(`Listening on port ${port}`));
 
 // Store and retrieve your videos from here
 // If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+//let videos = require("../client/src/exampleresponse.json");
+
+let connection = mysql.createConnection
+  ({
+    host: 'localhost',
+    database: 'sys',
+    user: 'root',
+    password: 'ArbeitsEmail123',
+    port: 3306
+  });
+
+//let result;
+
+
+
+let fail = {
+  "result": "failure",
+  "message": "Video could not be saved"
+};
 
 // GET "/"
-app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+app.get("/videos", (req, res) =>
+{
+  let sorted = req.query.order;
+
+  if (sorted === "asc")
+  {
+    connection.query('SELECT * FROM videos', (err, rows) =>
+    {
+      let result = Object.values(JSON.parse(JSON.stringify(rows)));
+      result = [...result].sort((a, b) => a.rating - b.rating);
+      res.send(result)
+    });
+    /*
+    videos = [...videos].sort((a, b) => a.rating - b.rating);
+    console.log(videos)
+    res.send(videos);*/
+  }
+
+  else if (sorted === "desc")
+  {
+    connection.query('SELECT * FROM videos', (err, rows) =>
+    {
+      result = Object.values(JSON.parse(JSON.stringify(rows)));
+      result = [...result].sort((a, b) => b.rating - a.rating);
+      res.send(result)
+    });
+    /*
+    videos = [...videos].sort((a, b) => b.rating - a.rating);
+    console.log(videos)
+    res.send(videos);
+    */
+  }
+
+  else
+  {
+    connection.query('SELECT * FROM videos', (err, rows) =>
+    {
+      result = Object.values(JSON.parse(JSON.stringify(rows)));
+      result = [...result].sort((a, b) => b.rating - a.rating);
+      res.send(result)
+    });
+  }
+
+});
+
+app.post("/videos", (req, res) =>
+{
+  const video = {
+    id: req.body.id,
+    title: req.body.title,
+    url: req.body.url,
+    rating: req.body.rating,
+    added: req.body.added
+  };
+
+  if (typeof req.body.title !== "string" || typeof req.body.url !== "string")
+  {
+    res.status(400).send(fail);
+  }
+
+
+  else
+  {
+    connection.connect(function (err)
+    {
+      let sql = `INSERT INTO videos 
+      (
+          id, title, url, rating
+      )
+      VALUES
+      (
+          ?, ?, ?, ?
+      )`;
+      connection.query(sql, [req.body.id, req.body.title, req.body.url, req.body.rating], (err, rows) =>
```

Comment:
not prone to sql injection, because we are splitting out the query from the values, so sql will protect you

Reply:
Noted, Thank You.